include/verilated_timing.h

```diff
@@ -148,14 +148,25 @@ class VlCoroutineHandle final {
 #endif
 };
 
+enum class VlDelayPhase : bool { ACTIVE, INACTIVE };
+
+struct VlDelayTime {
+    uint64_t time;
+    VlDelayPhase phase;
```

Comment:
Hmm.  This means every compare will need to take 2 loads and compares, which is unfortunate.  I wonder if it's faster overall to have two separate queues, and check both at unloading? Maybe benchmark it?  Will we have more then 2 phases eventually?

Reply:
Good catch. I made a new solution that utilizes a separate vector to store #0 delays.
A two-queues solutions yields a similar, marginally better performance that the original solution, but the one with vectors is noticeably better. I updated the PR with the new approach.

Btw, here's the code I wrote for benchmarking:
```systemverilog
module zero_delay_benchmark();
  initial begin
    repeat (1000) begin
      for (int i = 0; i < 10000; ++i) begin
        int x = 0;
        fork
          #1 ++x;
          #0 ++x;
        join_none
      end
    end
  end
endmodule
```

The number of iterations for inner and outer loops has been adjusted between ratios of 100k:100, 10k:1k and 1k:10k. Here are some of the results:

```
one_queue:
  100k:100:
    10.85user 0.94system 0:11.80elapsed 99%CPU
    10.38user 0.91system 0:11.31elapsed 99%CPU
    10.49user 0.96system 0:11.55elapsed 99%CPU
  10k:1k:
    10.33user 0.90system 0:11.24elapsed 99%CPU
    10.33user 0.91system 0:11.25elapsed 100%CPU
    10.37user 0.96system 0:11.34elapsed 99%CPU
  1k:10k:
    10.50user 0.97system 0:11.48elapsed 99%CPU
    10.35user 0.92system 0:11.28elapsed 100%CPU
    10.74user 0.94system 0:11.68elapsed 99%CPU
queue+vec:
  100k:100:
    6.14user 0.92system 0:07.07elapsed 99%CPU
    6.34user 0.88system 0:07.22elapsed 100%CPU
    6.17user 0.94system 0:07.11elapsed 99%CPU
  10k:1k:
    6.19user 0.88system 0:07.08elapsed 99%CPU
    6.16user 0.87system 0:07.09elapsed 99%CPU
    6.00user 0.88system 0:06.89elapsed 99%CPU
  1k:10k:
    6.18user 0.92system 0:07.11elapsed 99%CPU
    6.02user 0.87system 0:06.89elapsed 100%CPU
    6.34user 0.85system 0:07.20elapsed 100%CPU
 ```